src/server/game/Battlegrounds/BattlegroundMgr.cpp

```diff
@@ -334,6 +334,18 @@ Battleground* BattlegroundMgr::GetBattlegroundTemplate(BattlegroundTypeId bgType
     return bgs.empty() ? nullptr : bgs.begin()->second;
 }
 
+std::vector<const Battleground *> BattlegroundMgr::GetActiveBattlegrounds()
+{
+    std::vector<const Battleground *> result;
+
+    for (auto const& [bgType, bgData] : bgDataStore)
+        for (auto const& [id, bg] : bgData._Battlegrounds)
+            if (bg->GetStatus() == STATUS_WAIT_JOIN || bg->GetStatus() == STATUS_IN_PROGRESS)
+                result.push_back(static_cast<const Battleground*>(bg));
```

Comment:
```suggestion
                result.emplace_back(static_cast<const Battleground*>(bg));
```

Reply:
But what are the benefits of using emplace_back when dealing with raw pointers?